serverless/src/createConcepts/__tests__/handler.test.js

```diff
@@ -0,0 +1,181 @@
+      body: JSON.stringify({ message: 'Invalid input: RDF/XML data is required' }),
+      headers: {
+        ...mockDefaultHeaders,
+        'Content-Type': 'application/json'
+      }
+    })
+  })
+
+  test('should successfully create multiple concepts and return 200', async () => {
+    const mockRdfXml = `
+      <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:skos="http://www.w3.org/2004/02/skos/core#">
+        <skos:Concept rdf:about="concept1"/>
+        <skos:Concept rdf:about="concept2"/>
+      </rdf:RDF>
+    `
+    const event = { body: mockRdfXml }
+    sparqlRequest.mockResolvedValue({ ok: true })
+
+    const result = await createConcepts(event)
+
+    expect(sparqlRequest).toHaveBeenCalledWith({
+      contentType: 'application/rdf+xml',
+      accept: 'application/rdf+xml',
+      path: '/statements',
+      method: 'POST',
+      body: mockRdfXml
+    })
+
+    expect(result).toEqual({
+      statusCode: 200,
+      body: JSON.stringify({
+        message: 'Successfully loaded RDF XML into RDF4J',
+        conceptsLoaded: 2
+      }),
+      headers: {
+        ...mockDefaultHeaders,
+        'Content-Type': 'application/json'
+      }
+    })
+  })
+
+  test('should successfully create a single concept and return 200', async () => {
+    const mockRdfXml = `
+      <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:skos="http://www.w3.org/2004/02/skos/core#">
+        <skos:Concept rdf:about="concept1"/>
+      </rdf:RDF>
+    `
+    const event = { body: mockRdfXml }
+    sparqlRequest.mockResolvedValue({ ok: true })
+
+    const result = await createConcepts(event)
+
+    expect(result).toEqual({
+      statusCode: 200,
+      body: JSON.stringify({
+        message: 'Successfully loaded RDF XML into RDF4J',
+        conceptsLoaded: 1
+      }),
+      headers: {
+        ...mockDefaultHeaders,
+        'Content-Type': 'application/json'
+      }
+    })
+  })
+
+  test('should handle SPARQL endpoint errors', async () => {
+    const mockRdfXml = '<rdf:RDF><skos:Concept rdf:about="concept1"/></rdf:RDF>'
+    const event = { body: mockRdfXml }
+    sparqlRequest.mockResolvedValue({
+      ok: false,
+      status: 400,
+      text: () => Promise.resolve('Bad Request')
+    })
+
+    const result = await createConcepts(event)
+
+    expect(result).toEqual({
+      statusCode: 400,
+      body: JSON.stringify({
+        message: 'Error from SPARQL endpoint: Bad Request',
+        conceptsAttempted: 1
+      }),
+      headers: {
+        ...mockDefaultHeaders,
+        'Content-Type': 'application/json'
+      }
+    })
+  })
+
+  test('should handle unexpected errors', async () => {
+    const mockRdfXml = '<rdf:RDF><skos:Concept rdf:about="concept1"/></rdf:RDF>'
+    const event = { body: mockRdfXml }
+    sparqlRequest.mockRejectedValue(new Error('Network error'))
+
+    const result = await createConcepts(event)
+
+    expect(result).toEqual({
+      statusCode: 500,
```

Comment:
500 is normally for uncaught or unexpected errors. Is this one closer to "422 Unprocessable Content"?

Reply:
422 is for semantic errors, this is a catchall, i.e. if anything goes wrong with the request, it is a 500